include/sys/dbuf.h

```diff
@@ -127,8 +127,8 @@ typedef struct dbuf_dirty_record {
 	/* pointer back to our dbuf */
 	struct dmu_buf_impl *dr_dbuf;
 
-	/* pointer to next dirty record */
-	struct dbuf_dirty_record *dr_next;
+	/* list link for dbuf dirty records */
+	list_node_t dr_dirty_record_link;
```

Comment:
[nit] the `dr` stands for `dirty_record`, so `dr_dirty_record` is kind of redundant.  Also `list_node_t dr_dirty_node` uses the `_node` suffix vs this uses `_link`.  We should use the same suffix for both - I think `_node` is more commonly used in ZFS.

How about `dr_dbuf_node`, since it's a link on the dbuf's list?